website/src/pages/api/subscribe.ts

```diff
@@ -0,0 +1,63 @@
+// src/pages/api/subscribe.ts
+import { NextApiRequest, NextApiResponse } from 'next';
+import { createClient } from '@supabase/supabase-js';
+import { validateEmail } from '@/utils/validateEmail'; // Import the utility
+
+// Initialize Supabase with environment variables
+const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL!;
+const supabaseAnonKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!;
+const supabase = createClient(supabaseUrl, supabaseAnonKey);
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse,
+) {
+  if (req.method === 'POST') {
```

Comment:
**issue:** Avoid nested `if` here, it makes code less readable. Use early return instead.